src/printer.ts

```diff
@@ -1108,16 +1110,61 @@ export class PugPrinter {
 						case 'newline':
 							rawText += '\n';
 							break;
+						case 'interpolated-code':
+							usedInterpolatedCode = true;
+							rawText += tok.mustEscape ? '#' : '!';
+							rawText += `{${tok.val}}`;
+							break;
 						default:
-							logger.warn('Unhandled token for pipeless script tag:', JSON.stringify(tok));
+							logger.warn(
+								'[PugPrinter:start-pipeless-text]:',
+								'Unhandled token for pipeless script tag:',
+								JSON.stringify(tok)
+							);
 							break;
 					}
 
 					index++;
 					tok = this.tokens[index];
 				}
 
-				result = format(rawText, { parser, ...this.codeInterpolationOptions });
+				try {
+					result = format(rawText, { parser, ...this.codeInterpolationOptions });
+				} catch (error: unknown) {
+					if (usedInterpolatedCode) {
```

Comment:
I would probably invert this `if` to make it easier to distiguish when it's going to throw and when to warn.

````js
if (!usedInterpolatedCode) {
    [...]
    throw error;
}
// Continue without formatting the content
if (types.isNativeError(error)) {
[...]
````

Reply:
So I have restructured the warning context

The result looks something like this:  
![image](https://user-images.githubusercontent.com/7195563/101509405-06a16a80-3979-11eb-9b56-2e9ef7094b80.png)

Any further improvements?